PR: https://github.com/odoo/odoo/pull/282879

From: 62bce1329974dfc989110d2b0858203d7056ad33
From: adgu-odoo
Date: 2026-08-26 14:35:09

Breaking data model changes scores: del:2 + add:0, change matches:
-    number_of_rescue_session = fields.Integer(string="Number of Rescue Session", compute='_compute_current_session')
-    rescue = fields.Boolean(

Total Changes: 33

[IMP] point_of_sale: remove rescue session

Rescue sessions are not created anymore.
Before, they were use to store orders that were sync with sessions
after these sessions were closed.
Now if we try to sync an order with a closed session,
it will either be synced with the current open session (if any) or
raise an error if no valid session is available.
After that, the frontend will receive a notification from the backend
telling that the session is closed and will reload the page and then
resync the orders with the new session.
Rescue session is thus not needed anymore.

closes odoo/odoo#282879

Task-id: 6332359
Related: odoo/upgrade#11053
Signed-off-by: Adrien Guilliams (adgu) <adgu@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -128,7 +128,6 @@ class PosConfig(models.Model):
     session_ids = fields.One2many('pos.session', 'config_id', string='Sessions')
     current_session_id = fields.Many2one('pos.session', compute='_compute_current_session', string="Current Session", search='_search_current_session')
     current_session_state = fields.Char(compute='_compute_current_session')
-    number_of_rescue_session = fields.Integer(string="Number of Rescue Session", compute='_compute_current_session')
     current_cash_register_balance = fields.Float(compute='_compute_current_cash_register_balance', string="Cash Register")
     last_session_closing_date = fields.Date(compute='_compute_last_session')
     pos_session_username = fields.Char(compute='_compute_current_session_user')
@@ -372,13 +371,11 @@ class PosConfig(models.Model):
         self.session_ids.fetch(["state"])
         for pos_config in self:
             opened_sessions = pos_config.session_ids.filtered(lambda s: s.state != 'closed')
-            rescue_sessions = opened_sessions.filtered('rescue')
-            session = pos_config.session_ids.filtered(lambda s: s.state != 'closed' and not s.rescue)
+            session = pos_config.session_ids.filtered(lambda s: s.state != 'closed')
             # sessions ordered by id desc
             pos_config.has_active_session = bool(opened_sessions.filtered(lambda s: s.state != 'opening_control')) or False
             pos_config.current_session_id = (session and session[0].id) or False
             pos_config.current_session_state = (session and session[0].state) or False
-            pos_config.number_of_rescue_session = len(rescue_sessions)
 
     def _search_current_session(self, operator, value):
         if operator != 'in':
@@ -387,7 +384,7 @@ class PosConfig(models.Model):
 
     def _compute_statistics_for_session(self):
         for config in self:
-            session = config.session_ids.filtered(lambda s: s.state != 'closed' and not s.rescue)
+            session = config.session_ids.filtered(lambda s: s.state != 'closed')
             session_record = session[0] if session else None
             if not session_record or not session_record.exists():
                 config.statistics_for_current_session = False
@@ -539,7 +536,7 @@ class PosConfig(models.Model):
     @api.depends('session_ids')
     def _compute_current_session_user(self):
         for pos_config in self:
-            session = pos_config.session_ids.filtered(lambda s: s.state in ['opening_control', 'opened', 'closing_control'] and not s.rescue)
+            session = pos_config.session_ids.filtered(lambda s: s.state in ['opening_control', 'opened', 'closing_control'])
             if session:
                 pos_config.pos_session_username = session[0].user_id.sudo().name
                 pos_config.pos_session_state = session[0].state
@@ -1023,24 +1020,6 @@ class PosConfig(models.Model):
             'type': 'ir.actions.act_window',
         }
 
-    def open_opened_rescue_session_form(self):
-        rescue_session_ids = self.session_ids.filtered(lambda s: s.state != 'closed' and s.rescue)
-
-        if len(rescue_session_ids) == 1:
-            return {
-                'res_model': 'pos.session',
-                'view_mode': 'form',
-                'res_id': rescue_session_ids.id,
-                'type': 'ir.actions.act_window',
-            }
-        return {
-            'name': _('Rescue Sessions'),
-            'res_model': 'pos.session',
-            'view_mode': 'list,form',
-            'domain': [('id', 'in', rescue_session_ids.ids)],
-            'type': 'ir.actions.act_window',
-        }
-
     def get_limited_product_count(self):
         return self.env['ir.config_parameter'].sudo().get_int('point_of_sale.limited_product_count') or DEFAULT_LIMIT_LOAD_PRODUCT
 

--- a/addons/point_of_sale/models/pos_session.py
+++ b/addons/point_of_sale/models/pos_session.py
@@ -124,11 +124,6 @@ class PosSession(models.Model):
     )
     order_ids = fields.One2many('pos.order', 'session_id', string='Orders')
     order_count = fields.Integer(compute='_compute_order_count')
-    rescue = fields.Boolean(
-        string='Recovery Session',
-        help="Auto-generated session for orphan orders, ignored in constraints",
-        readonly=True,
-        copy=False)
     payment_method_ids = fields.Many2many(
         'pos.payment.method',
         related='config_id.payment_method_ids',
@@ -337,7 +332,6 @@ class PosSession(models.Model):
         if not onboarding_creation and self.search_count([
                 ('state', '!=', 'closed'),
                 ('config_id', '=', self.config_id.id),
-                ('rescue', '=', False),
             ]) > 1:
             raise ValidationError(_("Another session is already opened for this point of sale."))
 
